### PR TITLE
Add note about `params` hash in Action Controller Overview [ci skip] 

### PR DESCRIPTION
**Pull request by JunichiIto**
### Motivation / Background

Newbies tend to misunderstand `params` hash is a plain old Hash. So I added a note that `params` hash is ActionController::Parameters.

### Detail

This Pull Request changes "4 Parameters" in Action Controller Overview.

Screenshot 2024-01-18 at 10 31 55

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/source/action_controller_overview.md [hunk 1]
```diff
@@ -110,7 +110,10 @@ class ClientsController < ApplicationController
 end
 ```
 
+NOTE: `params` hash is not a plain old Hash; instead, it is [`ActionController::Parameters`][] object. While it behaves like Hash, it does not inherit from Hash class.
```
Comment (p8) [inline guides/source/action_controller_overview.md:113]: In the getting started guide we capitalize hash in "`params` Hash".

```suggestion
NOTE: The `params` Hash is not a plain old Ruby Hash; instead, it is an [`ActionController::Parameters`][] object. While it behaves like Hash, it does not inherit from Hash.
```

Comment (eugeneius) [reply guides/source/action_controller_overview.md:113]: There are five other examples of "`params` hash" in this guide already, though.

Comment (p8) [reply guides/source/action_controller_overview.md:113]: Ah, missed those while grepping. It seems "`params` hash" is more common.

**Review by zzak — APPROVED** [review]

File: guides/source/action_controller_overview.md (hunk 1 above)
Comment (p8) [reply guides/source/action_controller_overview.md:113]: @JunichiIto Could you add the missing articles (`The` and `an`)?

**Review by p8 — CHANGES_REQUESTED** [review]

File: guides/source/action_controller_overview.md (hunk 1 above)
Comment (JunichiIto) [reply guides/source/action_controller_overview.md]: @p8 I added your suggestion to batch. Is it okay?

Comment (p8) [reply guides/source/action_controller_overview.md]: Thanks @JunichiIto !
Can you squash your commits?

Comment (JunichiIto) [reply guides/source/action_controller_overview.md]: @p8 Sure. I squashed and pushed my commit.

**Comment by p8** [comment]
Thanks @JunichiIto !

**Pull request merged**